src/Sarif/Visitors/SuppressVisitor.cs

```diff
@@ -68,7 +72,18 @@ public override Result VisitResult(Result node)
                 suppression.SetProperty(nameof(expiryUtc), expiryUtc);
             }
 
-            node.Suppressions.Add(suppression);
+            if (this.resultsGuids != null && this.resultsGuids.Any()) 
+            {
+                if (this.resultsGuids.Contains(node.Guid))
```

Comment:
>Contains [](http://example.com/codeflow?start=38&length=8)

the default case-sensitive comparer is used if not specified.
should use a case-insensitive comparer here.